index.js

```diff
@@ -12,8 +12,16 @@ https://raw.githubusercontent.com/GPII/first-discovery-server/master/LICENSE.txt
 
 var fluid = require("infusion");
 var gpii = fluid.registerNamespace("gpii");
+var path = require("path");
 require("./src/js/firstDiscoveryServer.js");
+require("./src/js/config.js");
+
+var configFile = path.join(__dirname, "./fd_security_config.json");
+var ENVMap = gpii.firstDiscovery.server.config.oauth2.clientCredential.ENVMap;
+var schema = gpii.firstDiscovery.server.config.oauth2.clientCredential.schema;
+var oauth2Config = gpii.firstDiscovery.server.config.getConfig(configFile, ENVMap, schema);
 
 gpii.firstDiscovery.server({
-    port: process.env.FIRST_DISCOVERY_SERVER_TCP_PORT
+    port: process.env.FIRST_DISCOVERY_SERVER_TCP_PORT,
+    preferencesConfig: oauth2Config
```

Comment:
After the changes above, and the revision to the environment variable resolution scheme, this can just be a standard declarative config held in the project and can be started according to the standard Kettle init scheme (from an env var, command line argument, etc.)